File: S6/CAA/2599.py
memo = [[-1 for x in range(10**3)] for x in range(10**4)]

radares = []

def somatorio(pista, radar):
    if (pista-radar) <= 0 : return 1
    count = 0
    for x in radares:
        if memo[pista][x] == -1: memo[pista][x] = somatorio(pista-radar, x)
        count += memo[pista][x]
    return count
    # return sum([somatorio(pista-radar, x) for x in radares])

for teste in range(int(input())):
    n, m = list(map(int, input().split()))
    radares = list(map(int, input().split()))
    res = sum([somatorio(n, x) for x in radares])
    print(res%1000000007)

"""
Se o restante for 0, cobriu.

Botar um no final 

Tamanho da àra a ser coberta * numero de radares.
"""

File: S6/CAA/2599.cpp
#include <bits/stdc++.h>

using namespace std;

int memo[10010];

int radares[10010];

int a, r;

int f(int n) {
    if(n <= 0) return 1;
    if(memo[n] == -1) {
        memo[n] = 0;
        for(int i = 0; i < r; ++i) {
            memo[n] += f(n-radares[i]);
            memo[n] %= 1000000007;
        }
    }
}

int main()
{

    int t;
    cin >> t;
    while(t--){
        cin >> a >> r;
        for(int i = 0; i < r; i++)
            cin >> radares[i];
        memset(memo, -1, sizeof(memo));
        // int res = 0;
        // for(int i = 0; i < r; i++)
        //     res += f(a);
        cout << f(a) << endl;
    }
    
    return 0;
}
